cddl/measurement-map.cddl

```diff
@@ -1,5 +1,6 @@
 measurement-map = {
   ? &(mkey: 0) => $measured-element-type-choice
-  &(mval: 1) => measurement-values-map
+  ? &(mval: 1) => measurement-values-map
   ? &(authorized-by: 2) => [ + $crypto-key-type-choice ]
+  ? &(mkeyvalues: 3) => [ + mkeyvalue-pair ]
 }
```

Comment:
This encoding looks very clumsy.

If we are going to tweak the internals of the `measurement-map`, let's do itproperly :-)

In particular, it seems more natural to model it as:

```
measurement-map = {
  ? "__authorized-by" : [ + $crypto-key-type-choice ]
  + (tstr / uint) => measurement-values-map
}
```

1. `__authorized-by` needs to be a reserved keyword (note that it uses a [cut](https://datatracker.ietf.org/doc/html/rfc8610#section-3.5.4) in the map);
1. since both OIDs and UUIDs have canonical string representations, limiting the key types to `tstr` and `uint` doesn't look like a serious loss.)

Reply:
Where is the multiplicity gone in this modified  proposal?

Reply:
> Where is the multiplicity gone in this modified proposal?

`+ (tstr / uint) => measurement-values-map`

Reply:
Does that mean `measurement-values-map` has not fixed code point?

Reply:
> [...] since both OIDs and UUIDs have canonical string representations, limiting the key types to `tstr` and `uint` doesn't look like a serious loss.

That said, if we _really_ wanted we could define the key as an extensible type choice and do:
```
measurement-map = {
  "__authorized-by" : [ + $crypto-key-type-choice ]
  + $mkey-type-choice => measurement-value-map
}

$mkey-type-choice /= uint
$mkey-type-choice /= tstr
; $mkey-type-choice /= tagged-uuid ...
```

Not sure we'd have any strong reason for doing so; I just wanted to make it clear that the restriction in my proposal above is purely artificial.

Reply:
@deeglaze & @thomas-fossati : I am not convinced this is the right approach given the nature of complexity  for users to grasp, + lack of deterministic encoding for `measurement-values-map`.

Also we need the Mkey to be a bot more flexible, which in cases I am modelling could be a UUID or a UEID.

Let us discuss this in the next call, before fixating on this approach!